src/fileio.cpp

```diff
@@ -874,13 +874,14 @@ void CSphWriter::SeekTo ( SphOffset_t iPos, bool bTruncate )
 //////////////////////////////////////////////////////////////////////////
 
 static int g_iIOpsDelay = 0;
-static int g_iMaxIOSize = 0;
+static const int g_iLimitIOSize = ( 1UL << 30 ); // same as write chunk limit
+static int g_iMaxIOSize = g_iLimitIOSize; // by default, slice ios by at most 1 GB
 static std::atomic<int64_t> g_tmNextIOTime { 0 };
 
 void sphSetThrottling ( int iMaxIOps, int iMaxIOSize )
 {
 	g_iIOpsDelay = iMaxIOps ? 1000000 / iMaxIOps : iMaxIOps;
-	g_iMaxIOSize = iMaxIOSize;
+	g_iMaxIOSize = (0 < iMaxIOSize && iMaxIOSize < g_iLimitIOSize) ? iMaxIOSize : g_iLimitIOSize;
```

Comment:
We have 'Clamp' function. It performs the same, but looks more readable.
Clamp (min, max, v) returns Max ( min, Min ( max, v ) )

```
g_iMaxIOSize = Clamp (1, g_iLimitIOSize, iMaxIOSize);
```

Reply:
There was no such function in my forked (about 1 week ago) master. Now you have it. So, I'll try to use it

Reply:
done

Reply:
Actually I've identical wrong commit (9cb8f983088e98ebfcc0229730563208521df153) to my fork manticoresearch:master. I do not know how to remove it. Can you ignore it and use only alexiv1965:issue-2739-fix-io-chunk-size branch for PR?

Reply:
> Or provide me an advice how to remove it from PR, please.

@alexiv1965 Feel free to `git rebase` and `git push ... --force`. It's your branch, you are free to do whatever you want with it.